### ISSUE TITLE
3.0.9 release is sorting restore-keys alphabetically which dramatically changes behaviour

### ISSUE THREAD

**Issue by eastlondoner** [actions/cache#941]
This commit changes the restore-keys behaviour so that the order of the keys provided in the workflow is no longer respected when deciding what key to restore.

https://github.com/actions/cache/commit/204fa1ab69e643b91bf310c47cbc10fa2843d7fd#diff-79e9c65d7bc0c52a974cc3ca32be05e5fd06d175bfa85edbcef3af6df03ffe8dR150

This is a significant change in the behaviour and I think it's a bug

**Comment by vsvipul** [comment]
@eastlondoner Agreed. This slipped through and should have only been done with paths in the first place. I'll look into this.
